dobson/main.py

```diff
@@ -65,27 +68,115 @@ def respond_help(slack_client, channel):
     users = [device.user for device in utils.get_devices()]
+    unknown = list(utils.get_unknown_mac_addresses())
 
-    if len(users) > 2:
-        reply = '{}, and {} are at ozone'.format(', '.join(users[:-1]), users[-1])
-    elif len(users) == 2:
-        reply = '{} and {} are at ozone'.format(users[0], users[1])
-    elif len(users) == 1:
-        reply = '{} is at ozone'.format(users[0])
-    else:
-        reply = 'Nobody is at ozone'
+    final_response = UserResponse(users, unknown, message, list_all).random_response()
 
     slack_client.api_call(
         'chat.postMessage',
         channel=channel,
-        text=reply,
+        text=final_response,
         as_user=True,
     )
 
 
+class UserResponse:
+    """
+    A helper class to handle crafting response strings to dobson queries
+    """
+
+    def __init__(self, users, unknown, message, list_all):
+        self.users = users
+        self.unknown = unknown
+        self.message = message
+        self.list_all = list_all
+
+    def random_response(self):
+        """Returns a random response from our response methods"""
+        return random.choice([self.response1])()
+
+    def response1(self):
+        """
+        Sample responses:
+
+        Nobody is at ozone, but there is 1 unknown device...
+        Sean is at ozone, along with the following unknown devices: 000 and 111
+        """
+        # reply containing who is at ozone, e.g. "A, B and C are at ozone"
+        if self.users:
+            reply = '{list_formatted} {is_are} at ozone'.format(**UserResponse.grammar_helper(self.users))
+        else:
+            reply = 'Nobody is at ozone'
+
+        # Add unknown MAC Addresses to the response
+        if self.unknown:
+            unknown_prefix = ', ' + ('along with ' if self.users else ' but there {is_are} ')
+            if self.list_all:
+                unknown_reply = unknown_prefix + 'the following unknown device{s?}: {list_formatted}'
+                unknown_reply = unknown_reply.format(**UserResponse.grammar_helper(self.unknown))
+            else:
+                unknown_reply = unknown_prefix + '{num} unknown device{s?}{ellipsis}'
+                unknown_reply = unknown_reply.format(**UserResponse.grammar_helper(
+                    self.unknown,
+                    ellipsis='...' if not self.users else ''
+                ))
+        else:
+            unknown_reply = '.'
+
+        return reply + unknown_reply
+
+    @staticmethod
+    def grammar_helper(items, **kwargs):
+        """
+        Takes in a list of items and returns a dictionary mapping keywords to strings for use in format strings.
+
+        For example, if we pass in a list with 1 element, the key 'is_are' would be 'is', and a list with 2 or more
+        elements would be 'are'. This enables us to use format strings such as "[list of people] {is_are} home"
+
+        >>> "There {is_are} {num} device{s?} connected".format(**grammar_helper(['sean', 'jason', 'nikhil']))
```

Comment:
I feel like this method is a little bit overkill for what it does, I'd say something like just having a pluralizing method would be enough and would make things a bit simpler. All that method would do is to pluralize the thing passed into it depending on an argument saying how many items there are:

```
pluralize('person', 3)
>> 'people'
pluralize('person', 1)
>> 'person'
pluralize('person', 0)
>> 'people'
```

```
pluralize('is', 4)
>> 'are'
pluralize('is', 1)
>> 'is'
```

The difficulty here is more around irregular plurals, but you don't have to cover every case, just having a dictionary or something mapping plurals would be fine. I'm drawing inspiration for this from Rails, where there's an active_support `pluralize` method that does this, and it's quite useful but I'd say it's a bit simpler to use. Here's a decent example of a simpler version of it: https://stackoverflow.com/a/3030973

Thoughts? I'm happy to stick with the method you have here, since you've put a lot of time and work into it, and I'm not sure it's worth refactoring right now, but I feel like doing it in a simpler format would make it easier to read through, understand, maintain, and add new features to.

Reply:
Yeah, that makes sense. Doing something like that would probably be a lot simpler to understand. I'll keep it as is for now since it seems to work, but we can refactor it later.